server/src/main/java/org/elasticsearch/transport/InboundDecoder.java

```diff
@@ -203,23 +205,23 @@ private void ensureOpen() {
         }
     }
 
-    static IllegalStateException ensureVersionCompatibility(
-        TransportVersion remoteVersion,
-        TransportVersion currentVersion,
-        boolean isHandshake
-    ) {
-        // for handshakes we are compatible with N-2 since otherwise we can't figure out our initial version
-        // since we are compatible with N-1 and N+1 so we always send our minCompatVersion as the initial version in the
-        // handshake. This looks odd but it's required to establish the connection correctly we check for real compatibility
-        // once the connection is established
-        final TransportVersion compatibilityVersion = isHandshake ? currentVersion.calculateMinimumCompatVersion() : currentVersion;
-        if (remoteVersion.isCompatible(compatibilityVersion) == false) {
-            final TransportVersion minCompatibilityVersion = isHandshake
-                ? compatibilityVersion
-                : compatibilityVersion.calculateMinimumCompatVersion();
-            String msg = "Received " + (isHandshake ? "handshake " : "") + "message from unsupported version: [";
-            return new IllegalStateException(msg + remoteVersion + "] minimal compatible version is: [" + minCompatibilityVersion + "]");
+    static void checkHandshakeVersionCompatibility(TransportVersion handshakeVersion) {
+        if (handshakeVersion.before(TransportHandshaker.EARLIEST_HANDSHAKE_VERSION)) {
```

Comment:
I would prefer to check here that the handshake version is one of 6.8.0 or 7.17.0.

Reply:
Updated

Reply:
looks unchanged - did you forget to push?

Reply:
Yup :facepalm:

Reply:
Is 7.18.0 going to appear at any point now?

Reply:
There's no plan to do so. It would be very disruptive.